plutus-contract/src/Plutus/Contract/Test/ContractModel/Internal.hs

```diff
@@ -392,6 +398,12 @@ type SpecificationEmulatorTrace a =
 -- `ModelState` type, which in addition to @state@ tracks common features of the blockchain, like
 -- wallet balances and the current slot.
 
+class (Eq (Action state), Show (Action state)) => HasActions state where
+  getAllSymtokens :: Action state -> Set SymToken
+
+instance {-# OVERLAPPABLE #-} (Eq (Action state), Show (Action state), Data (Action state)) => HasActions state where
+  getAllSymtokens = Set.fromList . universeBi
```

Comment:
I looked into this (I didn't know about `DefaultSignatures`) and I don't see how they replace overlapping instances? As far as I can tell using this extension would require the programmer to state that they want an instance for `HasActions` for their datatype (possibly via `derive`?) whereas now you don't have to add this to your `deriving` list. Instead, if for some very specific reason you need to overwrite the default instance (because you can't rely on `Data` for whatever reason) you would now be required to add an `OVERLAPPING` pragma.

In other words: doing this would mean more work and more understanding in the common cases to have slightly less work (and very slightly less understanding) in the uncommon cases. I think I will pass on this suggestion here unless there is something I'm misunderstanding?

Reply:
There must be a downside to doing it this way, since I haven't seen this pattern before, have you? 
Maybe this didn't work well with the old `OverlappingInstances`?
I don't think it is a performance issue.
Or maybe it's just that in other cases it is more common to write you own instance and then having to add `OVERLAPPING` could be annoying.

Reply:
I think the last thing you wrote is most likely - with the new overlapping rules I don't see much room for confusion.ö